k8s/upgrade/src/upgrade_resources/upgrade.rs

```diff
@@ -84,22 +84,95 @@ pub struct GetUpgradeArgs {}
 impl GetUpgradeArgs {
     ///  Upgrade the resources.
     pub async fn get_upgrade(&self, namespace: &str) {
-        println!("{namespace}")
-        // To be implemented
+        // Create resources for getting upgrade status
+        let _res= GetUpgradeResource::create_get_upgrade_resource(namespace).await;
+    }
+}
+
+///  Resource to be created to get upgrade status.
+struct GetUpgradeResource {
+    upgrade_event: Api<Event>,
+}
+
+/// Methods implemented by GetUpgradeResource.
+impl GetUpgradeResource {
+    pub async fn new(ns: &str) -> Result<Self, Error> {
+        let client = Client::try_default().await?;
+        Ok(Self {
+            upgrade_event: Api::<Event>::namespaced(client, ns),
+        })
+    }
+
+    /// getter method to return upgrade event
+    pub(crate) fn get_upgrade_event(&self) -> Api<Event> {
+        self.upgrade_event.clone()
+    }
+
+    /// Fetch upgrade events
+    pub async fn get_events(&self) -> Result<(), kube::Error> {
```

Comment:
Because this is printing also, maybe we can name it as 'get_and_print_events" or just "print_events", or something more appropriate.